lib/spack/spack/cmd/create_tarball.py

```diff
@@ -43,11 +43,24 @@ def create_tarball(parser, args):
     if not args.packages:
         tty.die("tarball creation requires at least one package argument")
 
-    specs = spack.cmd.parse_specs(args.packages, concretize=True)
+    specs = spack.cmd.parse_specs(args.packages)
     for spec in specs:
-        package = spack.repo.get(spec)
+        # ----------- Make sure the spec only resolves to ONE thing
+        q = spack.installed_db.query(spec)
```

Comment:
spack create_tarball root
[root@6.07.02%clang@7.0.2-apple=darwin-x86_64^bzip2@1.0.6%clang@7.0.2-apple=darwin-x86_64^cmake@3.5.2%clang@7.0.2-apple~doc+ncurses+openssl~qt=darwin-x86_64^fftw@3.3.4%clang@7.0.2-apple+float+long_double~mpi~openmp~quad=darwin-x86_64^freetype@2.5.3%clang@7.0.2-apple=darwin-x86_64^ghostscript@9.16%clang@7.0.2-apple=darwin-x86_64^graphviz@2.38.0%clang@7.0.2-apple=darwin-x86_64^gsl@2.1%clang@7.0.2-apple=darwin-x86_64^jpeg@9a%clang@7.0.2-apple=darwin-x86_64^libxml2@2.9.2%clang@7.0.2-apple+python=darwin-x86_64^ncurses@6.0%clang@7.0.2-apple=darwin-x86_64^openssl@1.0.2f%clang@7.0.2-apple=darwin-x86_64^pcre@8.38%clang@7.0.2-apple=darwin-x86_64^python@2.7.11%clang@7.0.2-apple=darwin-x86_64^readline@6.3%clang@7.0.2-apple=darwin-x86_64^sqlite@3.8.5%clang@7.0.2-apple=darwin-x86_64^swig@3.0.8%clang@7.0.2-apple=darwin-x86_64^xz@5.2.2%clang@7.0.2-apple=darwin-x86_64^zlib@1.2.8%clang@7.0.2-apple=darwin-x86_64, root@6.07.02%gcc@4.9.3=darwin-x86_64^bzip2@1.0.6%gcc@4.9.3=darwin-x86_64^cmake@3.5.2%gcc@4.9.3~doc+ncurses+openssl~qt=darwin-x86_64^fftw@3.3.4%gcc@4.9.3+float+long_double~mpi~openmp~quad=darwin-x86_64^freetype@2.5.3%gcc@4.9.3=darwin-x86_64^ghostscript@9.16%gcc@4.9.3=darwin-x86_64^graphviz@2.38.0%gcc@4.9.3=darwin-x86_64^gsl@2.1%gcc@4.9.3=darwin-x86_64^jpeg@9a%gcc@4.9.3=darwin-x86_64^libxml2@2.9.2%gcc@4.9.3+python=darwin-x86_64^ncurses@6.0%gcc@4.9.3=darwin-x86_64^openssl@1.0.2f%gcc@4.9.3=darwin-x86_64^pcre@8.38%gcc@4.9.3=darwin-x86_64^python@2.7.11%gcc@4.9.3=darwin-x86_64^readline@6.3%gcc@4.9.3=darwin-x86_64^sqlite@3.8.5%gcc@4.9.3=darwin-x86_64^swig@3.0.2%gcc@4.9.3=darwin-x86_64^xz@5.2.2%gcc@4.9.3=darwin-x86_64^zlib@1.2.8%gcc@4.9.3=darwin-x86_64]
==> Error: Multiple matches for spec root.  Choose one:
root@6.07.02%clang@7.0.2-apple=darwin-x86_64
root@6.07.02%gcc@4.9.3=darwin-x86_64